jax/experimental/jax2tf/jax2tf.py

```diff
@@ -2566,9 +2568,16 @@ def _gather(operand, start_indices, *, dimension_numbers, slice_sizes: core.Shap
         indices_are_sorted=indices_are_sorted, fill_value=fill_value,
         output_shape=_out_aval.shape, _in_avals=_in_avals, _out_aval=_out_aval)
 
+  operand_aval = _in_avals[0]
   start_indices = _maybe_cast_to_int64(start_indices)
+  if core.is_opaque_dtype(operand_aval.dtype):
```

Comment:
WDYT about this simpler way of customizing the behavior for opaque types, instead of creating new _rules for jax2tf? Alternatively, we can create rules for adjusting the parameters of gather/slice/update_slice.

Reply:
This seems fine, and overall more consistent with thinking ahead to grounding out opaque dtypes within jaxpr (resolving jaxpr to physical-only jaxpr). If we can do that, then this sort of logic would actually just move up into the jaxpr-to-jaxpr rewrite instead. Would you agree?

I think having "rules" as we do currently would be fine if we only had one backend lowering target. With several targets, we end up with a set of rules for each, and a lot of conceptual redundancy.

Reply:
I agree. This deserves to be resolved in a distinct pass just before lowering.

There are other things that could fall in the same category, e.g., effects.